signals.c

```diff
@@ -0,0 +1,99 @@
+/*
+  Not part of Grbl. KeyMe specific.
+  
+  Signal is one layer of abstraction above adc.h and adc.c.
+
+  ADC values are read in specific time intervals, filtered and
+  stored in the appropriate arrays.
+*/
+
+#include "signals.h"
+#include "adc.h"
+#include "systick.h"
+
+#define N_FILTER 3
+#define SIGNALS_DEFAULT_INTERVAL 1000 
+#define X_BUF(i) adc_samples_x[FORCE_VALUE_INDEX][i]
+
+// Unfiltered ADC readings
+int16_t adc_samples_x[VOLTAGE_SENSOR_COUNT][N_FILTER]; 
+
+void signals_init()
+{
+  signals.pause = 0;
+  signals.callback_period = SIGNALS_DEFAULT_INTERVAL;
+  signals.force_offset = 0;
+}
+
+// Update motors ADC readings
+void signals_update_motors()
+{
+  uint8_t idx;
+
+  for (idx = 0; idx < N_AXIS; idx++) {
+    // Assumes the motors are on ADC channels 0-3 and in the same
+    // order in signals.adc_samples. If the pins are changed, a
+    // motors should be mapped to ADC channels.
+    signals.adc_samples[idx] = raw_adc_readings[idx];
+  }
+}
+
+// Filter and update force ADC reading
+void signals_update_force()
+{
+  // Since the raw adc value changes based on an interrupt,
+  // copy the raw value at the beginning of this function to
+  // ensure that the value being used doesn't change during 
+  // during the function call.
+  uint16_t raw_value = raw_adc_readings[FORCE];
+  
+  // Index in signal buffer
+  static const uint8_t n = N_FILTER - 1;
```

Comment:
Looks good.  I was actually going to suggest just having an aliasing define called something like 'SAMPLE', but this works too.